sweepbatcher/sweep_batch.go

```diff
@@ -316,7 +316,20 @@ func NewBatch(cfg batchConfig, bk batchKit) *batch {
 }
 
 // NewBatchFromDB creates a new batch that already existed in storage.
-func NewBatchFromDB(cfg batchConfig, bk batchKit) *batch {
+func NewBatchFromDB(cfg batchConfig, bk batchKit) (*batch, error) {
+	// Make sure the batch is not empty. We have checked this in
+	// spinUpBatchFromDB before calling NewBatchFromDB.
+	if len(bk.sweeps) == 0 {
```

Comment:
nit: I'd add a comment emphasizing that this should never happen as this precondition is already ensured in `spinUpBatchFromDB`.

Reply:
Fixed